### PR TITLE
refactor(openapi): move openapi to separate crate to decrease compile times

### PR DESCRIPTION
**Pull request by Narayanbhat166**
## Type of Change

- [x] Refactoring

## Description

Move All the openapi related changes to a separate crate to generate openapi files faster.

## How did you test it?

This is not a code related change. Only openapi related changes. In order to generate the openapi file, run the below command

```bash
cargo r -p openapi
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by sahkal — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

File: crates/euclid/src/frontend/ast.rs [hunk 1]
```diff
@@ -145,10 +149,44 @@ pub struct Rule<O> {
...
+    #[schema(value_type=HashMap<String, serde_json::Value>)]
     pub metadata: Metadata,
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
+pub struct RoutableConnectorChoice {
+    #[cfg(feature = "connector_choice_bcompat")]
+    #[serde(skip)]
+    pub choice_kind: RoutableChoiceKind,
+    // pub connector: RoutableConnectors,
```
Comment (SanchithHegde) [inline crates/euclid/src/frontend/ast.rs:168]: Remove commented code

File: crates/openapi/src/openapi.rs [hunk 2]
```diff
@@ -59,97 +60,132 @@ Never share your secret api keys. Keep them guarded and secure.
         (name = "Customers", description = "Create and manage customers"),
         (name = "Payment Methods", description = "Create and manage payment methods of customers"),
         (name = "Disputes", description = "Manage disputes"),
-        // (name = "API Key", description = "Create and manage API Keys"),
+        (name = "API Key", description = "Create and manage API Keys"),
```
Comment (SanchithHegde) [inline crates/openapi/src/openapi.rs:63]: Please include API keys endpoints in the docs as well.

**Review by SanchithHegde — APPROVED** [review]

**Review by sai-harsha-vardhan — DISMISSED** [review]

**Review by sahkal — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Review by sahkal — APPROVED** [review]

**Comment by Narayanbhat166** [comment]
No test cases for this PR as only open api related changes are included in this

**Review by vspecky — APPROVED** [review]

**Comment by vspecky** [comment]
@Narayanbhat166 Can we generate mintlify docs for this locally and just verify it all looks good?

**Comment by Narayanbhat166** [comment]
Yes, the mintlify docs can be verified at this url https://juspay-29.mintlify.app/api-reference/merchant-account/merchant-account--create

**Pull request merged**